### ISSUE TITLE
Dihydroorotate Dehydrogenase

### ISSUE THREAD

**Issue by Devlin-Moyer** [SysBioChalmers/Human-GEM#609]
DHODH (`ENSG00000102967`) catalyzes `MAR04575`: `(S)-dihydroorotate [c] + ubiquinone [m] ⇔ orotate [c] + ubiquinol [m]`

While DHODH is capable of oxidizing dihydroorotate to orotate and reducing ubiquinone to ubiquinol, it uses FMN (`MAM01828c/e`) as an intermediate electron carrier in between those two substrates ([source](https://febs.onlinelibrary.wiley.com/doi/full/10.1111/febs.16164)). Most reactions that are catalyzed by similar enzymes that use FAD in the same way, speifically those catalyzed by some combination of ACADVL, ACADL, ACADM, ACADS, and ACADSB (that's over 100 reactions in total, so I'm not listing them out here) do not directly involve ubiquinone (`MAM03103m`), but instead oxidize/reduce FAD(H<sub>2</sub>) (`MAM01802m`/`MAM01803m`), which oxdizes/reduces ubiquinone/ubiquinol in `MAR06911`. To be consistent with the representation of the existing reactions catalyzed by FAD-dependent enzymes where the electrons eventually wind up reducing ubiquinone:

- [ ] ~Change ID of `MAM20022i` to `MAM20019i` to be consistent with `MAM20019r`, since both metabolite objects are supposed to represent the same compound (FMNH<sub>2</sub>)~ addressed by #712
- [X] Create new metabolite `MAM20019c` to represent cytosolic FMNH<sub>2</sub>
- [X] Replace `MAM03103m` (ubiquinone) with `MAM01828c` (FMN) and `MAM03102m` (ubiquinol) with `MAM20019c` (FMNH<sub>2</sub>) in `MAR04575`
- [X] Create a new reaction to represent redox of FMN(H<sub>2</sub>) and ubiquinone/ubiquinol catalyzed by DHODH: `MAM20019c + MAM03103m <-> MAM01828c + MAM03102m`, GPR: `ENSG00000102967`, references: `PMID:34428349`

Technically, the redox between FMN(H<sub>2</sub>) and ubiquinone/ubiquinol happens at the outer face of the inner mitochondrial membrane, but there isn't currently a mitochondrial FMN metabolite, and it seems like overkill to add mitochondrial versions of both FMN and FMNH<sub>2</sub> as well as cytosol <-> mitochondria transport reactions for both.

There are some other FMN-dependent enzymes associated with reactions in Human1 that do not currently involve the FMN metabolite (e.g. [PNPO](https://metabolicatlas.org/explore/Human-GEM/gem-browser/gene/ENSG00000108439), see Table 1 of [this paper](https://febs.onlinelibrary.wiley.com/doi/full/10.1111/febs.16164)) that could probably have similar edits made to them, but I came across this issue when specifically looking into enzymes that use FAD to reduce ubiquinone, so I haven't looked into exactly what changes would need to be made to those reactions (yet).

**Comment by JonathanRob** [comment]
Seems like a reasonable suggestion, since as you say many FAD-binding enzymes are already represented this way in the model. My concerns are that by separating the reaction steps, it becomes a bit less clear what the overall reaction is because one needs to be aware of both steps. Also, if there are other reactions that can reduce/oxidize FMN, it opens the possibility to use those as the second step instead of coupling to ubiquinone/ubiquinol (I know there aren't any now because it's a new metabolite, but if some are added in the future).

For the new reaction step involving electron transfer between FMNH2 and ubiquinone, would the GPR of that reaction include all FMN-dependent enzymes?

**Comment by Devlin-Moyer** [comment]
> My concerns are that by separating the reaction steps, it becomes a bit less clear what the overall reaction is because one needs to be aware of both steps. Also, if there are other reactions that can reduce/oxidize FMN, it opens the possibility to use those as the second step instead of coupling to ubiquinone/ubiquinol (I know there aren't any now because it's a new metabolite, but if some are added in the future).

So while I agree that it does make the overall reaction less clear, that may actually (unfortunately) be a more accurate representation of reality, since no FMN-dependent flavoenzymes in humans bind their FMN covalently (a few of the FAD-dependent ones do, but the vast majority don't) [source](https://doi.org/10.1016/j.abb.2013.02.015), and at least one human enzyme that had long been thought to depend on FAD and/or FMN bound to a different enzyme was shown to _preferentially_ use *free* FAD and/or FMN to catalyze its redox reaction [source](https://doi.org/10.1074/jbc.M109.059485)

> For the new reaction step involving electron transfer between FMNH2 and ubiquinone, would the GPR of that reaction include all FMN-dependent enzymes?

As far as I can tell, it would only contain DHODH, because that seems to be the only human FMN-dependent enzyme that is known to be capable of directly transferring electrons between its FMN and ubiquinone

**Comment by JonathanRob** [comment]
Sounds great, thanks for the detailed answers @Devlin-Moyer!

**Comment by Devlin-Moyer** [comment]
I was just about to start working on this when I noticed that there are already two metabolites in Human-GEM called "FMNH2": `MAM20019r` and `MAM20022i`. While I'd still need to add cytosolic and mitochondrial FMNH<sub>2</sub> metabolites for this issue, I get the impression that all compartment-specific versions of the "same" metabolite are supposed to have identical numeric bits of their IDs and only differ in their final characters that specify the compartments. Should I also change the ID of `MAM20022i` to `MAM20019i` for consistency with `MAM20019r` as part of the pull request for this issue (and then make the new FMNH<sub>2</sub> metabolites for this issue be `MAM20019c/m`), or should I make that a separate issue?

(I verified that none of the proposed changes in #610, #617, #618, #620, #621, or #622 would address this specific pair of metabolites)
